### DISCUSSION TITLE
Difference between dvhe.08.06 and dvhe.08.07?

### DISCUSSION THREAD

**Discussion by manuelrn** [quietvoid/dovi_tool#103] [General]
Hi!

What is the difference between `dvhe.08.06` and `dvhe.08.07`?
I always imagined that it was the resolution, because when I mux a 1080 video with MKVToolNix the MediaInfo shows `dvhe.08.03`, but if I mux a 2160 video the MediaInfo shows `dvhe.08.06`.

This has always happened to me, so I have always thought that 1080 is `dvhe.08.03` and 2160 is `dvhe.08.06`.
But now I just mux a 2160 video and the MediaInfo shows `dvhe.08.07` (instead of `dvhe.08.06` as it had always happened to me).

So what is the difference? Why was it `dvhe.08.07` this time?
And is `dvhe.08.07` correct? Or is `dvhe.08.07` wrong and it should be `dvhe.08.06` and the bug should be reported to the MKVToolNix maintainer?
<br/>

Thanks and regards!

**Comment by quietvoid** [comment]
That number is the level.
For MKVToolNix, it's calculated from the resolution and frame rate of the video.
Are the videos differing in either of those?

**Reply by manuelrn** [reply]
> That number is the level. For MKVToolNix, it's calculated from the resolution and frame rate of the video. Are the videos differing in either of those?

Ah, I did not know that the frame rate also influences. So I guess it must be the frame rate.

The content that I usually mux is usually 23.976 FPS or 24 FPS, with them the MediaInfo has always shown `dvhe.08.06` for 2160p and `dvhe.08.03` for 1080p.
But with this content I was worried that the MediaInfo shows `dvhe.08.07` for 2160p and `dvhe.08.04` for 1080p... But I just checked and this content is 25 FPS, so that's why it must be different, right?

Thanks again!

**Reply by quietvoid** [reply]
Yes. The max for level 6 is 3840x2160 at 24 fps.
You can look at the code responsible here https://gitlab.com/mbunkus/mkvtoolnix/-/blob/main/src/common/dovi_meta.cpp#L196

**Reply by manuelrn** [reply]
> Yes. The max for level 6 is 3840x2160 at 24 fps. You can look at the code responsible here https://gitlab.com/mbunkus/mkvtoolnix/-/blob/main/src/common/dovi_meta.cpp#L196

Ok, my question is resolved. Thanks again for your help!

**Comment by ghost** [comment]
The bitrate also influences which level should be selected, which mkvtoolnix doesn't take into account.  Their spec chart on the levels really only cares about pixel rate, picture width, and bitrate.   The resolutions are just given as examples.  My old Galaxy Note8 (combined with OpenCamera at least) could record 1080p at insanely high bitrate.   I'm looking at one right now that shows up as VFR ~9-30fps 147Mb/s Main@5.2.  As Main10 that would require level 13 DV.  That phone only shot 8 bit HEVC so it wouldn't be usable without re-encoding anyway, but I'd bet on a re-encode at low CRF to a high profile still topping the 70Mb/s limit of the 1080p resolutions mkvtoolnix would calculate.  

I have no use-case for dovi in MKV and no gitlab account but someone who does may want to bring this up over there.

**Reply by quietvoid** [reply]
Bitrate isn't something simple to add, see https://gitlab.com/mbunkus/mkvtoolnix/-/issues/3113#note_706379247

**Comment by cromefire** [comment]
Is there anyway to convert that when it is 4K@24 and dvhe.08.07? mkvtoolnix doesn't seem to do it anymore and I have some files with dvhe.08.07 that won't play (just get a solid purple screen), while ones with dvhe.08.06 will and that's basically the only difference between them at least from what mediainfo or `dovi_tool info` can find out, besides the second one having HDR10+ (HDR10+ at least works fine all the time, which you can't say of DV...).

works:
```
Format                                   : HEVC
Format/Info                              : High Efficiency Video Coding
Format profile                           : Main 10@L5@High
HDR format                               : Dolby Vision, Version 1.0, dvhe.08.06, BL+RPU, HDR10 compatible / SMPTE ST 2086, HDR10 compatible
Codec ID                                 : V_MPEGH/ISO/HEVC
Bit rate                                 : 23 Mb/s
Width                                    : 3 840 pixels
Height                                   : 1 920 pixels
Display aspect ratio                     : 2.000
Frame rate mode                          : Constant
Frame rate                               : 23.976 (24000/1001) FPS
Color space                              : YUV
Chroma subsampling                       : 4:2:0 (Type 2)
Bit depth                                : 10 bits
```

doesn't work:
```
Format/Info                              : High Efficiency Video Coding
Format profile                           : Main 10@L5@High
HDR format                               : Dolby Vision, Version 1.0, dvhe.08.07, BL+RPU, HDR10 compatible / SMPTE ST 2094 App 4, Version 1, HDR10+ Profile B compatible
Codec ID                                 : V_MPEGH/ISO/HEVC
Bit rate                                 : 24 Mb/s
Width                                    : 3 840 pixels
Height                                   : 2 160 pixels
Display aspect ratio                     : 16:9
Frame rate mode                          : Constant
Frame rate                               : 23.976 FPS
Color space                              : YUV
Chroma subsampling                       : 4:2:0 (Type 2)
Bit depth                                : 10 bits
```

Even tried to convert with mode 0 and 2, but that didn't help at all. The only thing that does help is to remove DV, which then makes it fall back to HDR10+ (or remove HDR10+, then DV works, see below), which just works. My suspicion is that my TV doesn't like it (e.g. doesn't support level 7) because it only does DV up to 4K@24 I think, not like LG which supports it for gaming as well for example (so up to 120Hz I think).

So funnily enough, while typing this comment I found out it also works if I strip HDR10+, then it results in dvhe.08.06 even when I readd HDR10+ it beaks again, but I have other files combining both where it works absolutely fine:
```
Format                                   : HEVC
Format/Info                              : High Efficiency Video Coding
Format profile                           : Main 10@L5@Main
HDR format                               : Dolby Vision, Version 1.0, dvhe.08.06, BL+RPU, HDR10 compatible / SMPTE ST 2094 App 4, Version 1, HDR10+ Profile A compatible
Codec ID                                 : V_MPEGH/ISO/HEVC
Bit rate                                 : 19 Mb/s
Width                                    : 3 840 pixels
Height                                   : 1 600 pixels
Display aspect ratio                     : 2.40:1
Frame rate mode                          : Constant
Frame rate                               : 23.976 (24000/1001) FPS
Color space                              : YUV
Chroma subsampling                       : 4:2:0
Bit depth                                : 10 bits
```

Maybe I'm looking at the wrong differences here but I can't find out what's the issue here... It seems like that they are the same, besides the profile level 7, but somehow not.

**Reply by quietvoid** [reply]
The difference is that your level 6 files are not 16:9 thus less pixels.
Everything should be playing 2160p24 files though.

**Reply by cromefire** [reply]
You're right, shouldn't make any difference though, but I might as well try and crop them... On the other hand I also had a version of the DV+HDR10+ file which was 2160p and worked. Maybe I need to search through my files for some other examples.
